taichi/ir/statements.h

```diff
@@ -275,6 +310,23 @@ class ThreadLocalPtrStmt : public Stmt {
   TI_DEFINE_ACCEPT_AND_CLONE
 };
 
+class BlockLocalPtrStmt : public Stmt {
+ public:
+  Stmt *offset;

```

Comment:
Another type of pointer...

TODO: better `maybe_same_address`, and something like `definitely_same_address` to replace some abuse of `irpass::analysis::same_statements` in `control_flow_graph.cpp`. 

`definitely_same_address` for `BlockLocalPtrStmt` can be `irpass::analysis::same_statements(a->offset, b->offset)`, which is more accurate than `irpass::analysis::same_statements(a, b)`.